PricingLoader.scala

```diff
@@ -0,0 +1,54 @@
+package org.llm4s
+
+import scala.io.Source
+import scala.util.{Try, Using}
+import play.api.libs.json._
+
+object PricingLoader {
+  private val pricingUrl = "https://raw.githubusercontent.com/BerriAI/litellm/main/model_prices_and_context_window.json"
+  private var cachedPricing: Option[JsValue] = None

```

Comment:
We should create a Case class to represent each entry - there is a lot of useful information within the mappings table - the context window lengths and other flags available are really really useful.

Reply:
And rather then we can create a Map[(privider, model)]: CostInfomation